### PR TITLE
Spreadsheet column

### PR DESCRIPTION
**Pull request by JBRPG**
Successfully implemented the add columns for the spreadsheet. Also made the add fields file more modular.

### CODE REVIEW

**Comment by JBRPG** [comment]
Add field function has been improved to become modular so it can work with all views, including spreadsheet and map for create form. Did the best I could to cut down the many obvious bugs during the implementation.

**Comment by JBRPG** [comment]
I felt proud to finally to submit a pull request filled with significant improvement to adding fields through forms and spreadsheet. Also revised the error handling system by simplifying the HTML output and having a dedicated collection of strings to store and give out the appropriate message.

I also managed to get most of the tests updated to reflect the changes. Even if I ended up having a few tests have confusing errors due to the changes made to the structure quite some time ago, I still feel proud that I put effort in updating existing tests to accommodate the changes from the actual files.

I do not know when I will be returning to the tests to finish off the errors or whatnot, but the important thing is that I managed to make the web browsing experience better for the end-user.

**Comment by vanwars** [comment]
A few things to fix:

* Please remove all console.log statements
* In the "Create Form" UI, the delete field/row is not working. Please fix it and add a test that checks to make sure that it's working.

File: static/views/create-form.js [hunk 1]
```diff
@@ -110,23 +117,30 @@ define([
...
+            if (!fieldsValidated) {
+                this.render();
+                return;
+            }
 
             this.model.set('name', formName);
             this.model.set('caption', caption);
             this.model.set('slug', 'slug_' + parseInt(Math.random() * 100000, 10));
             this.model.set('project_ids', [this.app.getProjectID()]);
+            // Some way, there has to be a condition to
```
Comment (vanwars) [inline static/views/create-form.js]: can you delete this comment?

File: static/tests/spec/views/spreadsheet-test.js [hunk 2]
```diff
@@ -179,7 +179,7 @@ define([
                 });
 
                 it("Successfully calls attachModels", function(){
-                    // Unforntunately, this one requires digging into oen of the elements
+                    // Unforntunately, this one requires digging into one of the elements
```
Comment (vanwars) [inline static/tests/spec/views/spreadsheet-test.js]: Can you delete this comment?

File: static/tests/spec/views/create-form-test.js [hunk 3]
```diff
@@ -263,6 +277,78 @@ define([
...
+
+                it("Successfully passes all validation tests", function() {
+                    newCreateForm = new CreateForm({
+                        app: this.app,
+                        model:new Form({id: 5})
+                    });
+                    newCreateForm.render();
+                    fixture = setFixtures("<div></div>").append(newCreateForm.$el);
+                    expect(CreateForm.prototype.addFieldButton).toHaveBeenCalledTimes(0);
+
```
Comment (vanwars) [inline static/tests/spec/views/create-form-test.js]: Can you delete this comment?

File: static/tests/spec/views/create-form-test.js [hunk 4]
```diff
@@ -193,6 +204,8 @@ define([
                     fixture.find('.new_field_button').trigger('click');
                     expect(CreateForm.prototype.addFieldButton).toHaveBeenCalledTimes(1);
```
Comment (vanwars) [inline static/tests/spec/views/create-form-test.js]: Can you delete this comment?

File: static/tests/spec/views/create-form-test.js [hunk 5]
```diff
@@ -73,17 +83,19 @@ define([
 
         describe("Create Form: Attach Collection Event Handler", function () {
             beforeEach(function () {
-                initSpies();
+                initSpies(this);
                 newCreateForm = new CreateForm({
                     model: this.form
                 });
```
Comment (vanwars) [inline static/tests/spec/views/create-form-test.js]: Can you delete this comment?

**Pull request merged**